PowerEditor/src/Notepad_plus.rc

```diff
@@ -309,6 +309,7 @@ BEGIN
         BEGIN
             MENUITEM "Duplicate Current Line",                           IDM_EDIT_DUP_LINE
             MENUITEM "Remove Consecutive Duplicate Lines",               IDM_EDIT_REMOVE_DUP_LINES
+            MENUITEM "Remove Any Duplicate Lines, Keeping First Occurrence", IDM_EDIT_REMOVE_ANY_DUP_LINES

```

Comment:
I was keeping the door open for adding a `..., Keeping Last Occurrence` option if there is a demand for it. :-)
If we want to keep that door open, but still be shorter, how about `Remove Any Duplicate Lines, Keeping First`?

Reply:
Alternatively, since it is already on the _Line Operations_ menu, _Remove Any Duplicates, Keeping First_ might also be an option.

Reply:
"Remove Any Duplicate Lines, Keeping First" still too long.
However, we can change "Remove Consecutive Duplicate Lines" to underline the difference of these 2 commands.
Any idea?

Reply:
Also, "Keeping First" idea is implicated, so adding these words is not necessary.

Reply:
`Remove Adjacent Duplicate Lines` << shorter than Consecutive 
`Remove Non-First Duplicate Lines` << the information about keeping this first duplicate is important here, it would be strange to inform about it only in the manual. 
@sasumner is native, maybe he'll come up with something shorter/better.

What is the acceptable limit to the command length? I see some others are pretty long too:
```
Remove Empty Lines (Containing Blank characters)
Sort Lines As Decimals (Comma) Descending

Remove Any Duplicate Lines, Keeping First << not so much:)
```

Reply:
>However, we can change "Remove Consecutive Duplicate Lines" to underline the difference of these 2 commands.

>Remove Adjacent Duplicate Lines << shorter than Consecutive

You know I don't mess with established stuff -- that's why we make a good decision from the start! :-)

>Remove Non-First Duplicate Lines

Sorry. Ouch. Painful. :-)

>sasumner is native, maybe he'll come up with something shorter/better.

I gave it my best shot on the first try.
I like what I originally proposed, and I still like it if I drop `Occurrence`.
But...English expression:  This is @donho 's football, so I'm going to change it to the suggested, simple `Remove Duplicate Lines`

BTW, with a code change to this PR of only one line, it could do the "Keeping Last" version of this command.  I have seen users request this, but not frequently.
So I was going to leave that out, but reserve for (it in a way) by using "Keeping First" in the menu name.  That's all I was after here.

Reply:
>Also, "Keeping First" idea is implicated

In fact, we can assume this because both of these commands keep the first occurrence:
```
Remove Duplicate Lines
Remove Consecutive Duplicate Lines
```

Of course, until a request for "Keeping Last" not appear.

Reply:
>because both of these commands keep the first occurrence

How do you know `Remove Consecutive Duplicate Lines` keeps the _first_ occurrence??
Maybe it keeps the _last_ !?
(You can't tell, because either way the effect is exactly the same)
Not so for the command introduced by this PR (that's why there could be 2 versions of it).

>"Remove all"

I thought about this, but I have never seen a user request for it.
To be clear: a strict definition of "Remove All" would mean that if a line `abc` is seen two or more times in the document/range, after the command NO `abc` lines would exist.
Perhaps that is a downside to @donho 's suggested text -- it implies the command would do what I just explained

Reply:
>How do you know Remove Consecutive Duplicate Lines keeps the first occurrence??
Maybe it keeps the last !?
(You can't tell, because either way the effect is exactly the same)

Because better fit to first occurrence, data is the same so I check on which line the data is remained (always in the first). Related to "Remove all" in other editor I also see something like "Keep only unique lines".

Even `Remove Consecutive Duplicate Lines` doesn't explicitly indicate whether the first occurrence remains or is deleted. So `Remove Duplicate Lines` it's not that bad (although at the beginning it was also unclear to me, as seen above).

Reply:
> BTW, with a code change to this PR of only one line, it could do the "Keeping Last" version of this command. 

Then "Keep the nth line"... No, please. I like KISS and you should do as well :) 

> What is the acceptable limit to the command length? I see some others are pretty long too

It's not the reason to make long line command. With such thinking the command will get longer & longer.


So I am wondering if it's better : 
```
Remove All Duplicate Lines
Remove Consecutive Duplicate Lines
```
If you prefer just 
```
Remove Duplicate Lines
Remove Consecutive Duplicate Lines
```
I'm OK too. We can always change them if users complain.

Please move new command `Remove (All) Duplicate Lines` before `Remove Consecutive Duplicate Lines`

Reply:
@donho said:

>I like KISS and you should do as well :)

This is interesting because I've been accused of writing "over-complicated" code in the past. 
This was said by people who didn't code "enough" to solve the problem completely. 
I like to think that my code is just "complicated enough" to accomplish the task well. :-)

Reply:
>If you prefer just... `Remove Duplicate Lines`

I'm going to use exactly that.
In English at least, the addition of `All` implies the "Keep Only Unique Lines" functionality.

Reply:
BTW, for future reference, I will add comments (to the linked "issue") to show how the code can be changed to create the following, in case wanted in the future:

* Remove Duplicate Lines (Keep Last)
* Keep Only Unique Lines